engine/classes/ElggWidgetDefinition.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/**
+ * \ElggWidgetDefinition
+ *
+ * Helper class for defining a widget
+ *
+ * @package    Elgg.Core
+ * @subpackage Widgets
+ *
+ */
+class ElggWidgetDefinition {
+
+	/**
+	 * @var string Handler of the widget
+	 */
+	public $handler;
+	
+	/**
+	 * \ElggWidgetDefinition constructor
+	 *
+	 * @param string $handler Handler of the widget
```

Comment:
only problem is handler is currently expected when data is returned in elgg_get_widget_types, so should we rename to id internally, but provide ->handler via a magic __get?? or just keep handler to keep everything inline??